Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/IndirectQuickRun.py

```diff
@@ -220,58 +205,204 @@ def _setup(self):
+
+    def _perform_two_peak_fits(self, workspace_names, x_min, x_max):
+        result_workspaces = []
+        chi_workspaces = []
+        run_numbers = []
+        for workspace_name in workspace_names:
+            number_of_histograms = mtd[workspace_name].getNumberHistograms()
+
+            progress_tracker = Progress(self, 0.3, 1.0, number_of_histograms+1)
+            progress_tracker.report('Finding temperature for {0}...'.format(workspace_name))
+
+            if self._temperatures is None:
+                run_number = str(mtd[workspace_name].getRunNumber())
+                run_numbers.append(run_number)
+
+            result = workspace_name[:-3] + 'fit'
+
+            for index in range(number_of_histograms):
+                progress_tracker.report('Fitting {0}-sp{1}...'.format(workspace_name, index))
+                IndirectTwoPeakFit(SampleWorkspace=workspace_name,
+                                   EnergyMin=x_min,
+                                   EnergyMax=x_max,
+                                   OutputName=result)
+
+            result_workspaces.append(result + '_Result')
+            chi_workspaces.append(result + '_ChiSq')
+        return result_workspaces, chi_workspaces, run_numbers
+
+    def _create_width_workspace(self, result_workspaces, temperatures, run_numbers, output_name, x_axis_is_temperature):
+        self._extract(result_workspaces[0], 0, output_name)
+        for index, workspace in enumerate(result_workspaces[1:]):
+            self._extract(workspace, 0, '__spectrum')
+            self._append(output_name, '__spectrum', output_name)
+
+        DeleteWorkspace(Workspace='__spectrum')
+        self._format_width_workspace(mtd[output_name], temperatures, run_numbers, x_axis_is_temperature)
+
+    @staticmethod
+    def _format_width_workspace(width_workspace, temperatures, run_numbers, x_axis_is_temperature):
+        number_of_temperatures = len(temperatures)
+        axis = NumericAxis.create(number_of_temperatures)
+        for index in range(number_of_temperatures):
+            value = float(temperatures[index]) if x_axis_is_temperature else float(run_numbers[index][-3:])
+            axis.setValue(index, value)
+        width_workspace.replaceAxis(1, axis)
+        width_workspace.setYUnitLabel("Temperature")
+
+    def _create_diffusion_workspace(self, width_workspace, temperatures, run_numbers, output_name, x_axis_is_temperature):
+        x_data = []
+        y_data = []
+        e_data = []
+        for index in range(width_workspace.getNumberHistograms()):
+            x = width_workspace.readX(index)
+            y = width_workspace.readY(index)
+            e = width_workspace.readE(index)
+            x_data.append(float(temperatures[index])) if x_axis_is_temperature else x_data.append(float(run_numbers[index][-3:]))
+            y_data.append(y[5] / x[5])
+            e_data.append(e[5] / x[5])
```

Comment:
we just he 5th data point?

Reply:
@AnthonyLim23 Width is proportional to Q in the limit of low Q. A proper fit is done in FqFit where an appropriate fitting range can be selected. For our purposes here it is sufficient & quick  to just use a single point. The first can be unreliable, so 5 is my guess for a suitable value. and shouldn't be too big.

Reply:
again could we just have a comment explaining it?